RIOT-OS/RIOT
# boards: model nrf52 boards

### Contribution description

Follow up to https://github.com/RIOT-OS/RIOT/pull/16837 this models the remaining BOARDs
I had to do multiple changes to model this one, so commits are better checked one by one.

### Testing procedure

- Gren CI
- Check that driver modeling still makes sense
- Check that default bootloader config makes sense

### Issues/PRs references

part of #16875
